coredns/datadog_checks/coredns/coredns.py

```diff
@@ -0,0 +1,110 @@
+            # Note: the count metrics were moved to specific functions
+            # below to be submitted as both gauges and monotonic_counts
+            # 'coredns_dns_response_rcode_count_total': self.coredns_dns_response_rcode_count_total,
+            # 'coredns_proxy_request_count_total': self.coredns_proxy_request_count_total,
+            # 'coredns_cache_hits_total': self.coredns_cache_hits_total,
+            # 'coredns_cache_misses_total': self.coredns_cache_misses_total,
+            # 'coredns_dns_request_count_total': self.coredns_dns_request_count_total,
+            # 'coredns_dns_request_type_count_total': self.coredns_dns_request_type_count_total,
+        }
+
+        # Create instances we can use in OpenMetricsBaseCheck
+        generic_instances = None
+        if instances is not None:
+            generic_instances = self.create_generic_instances(instances)
+
+        super(CoreDNSCheck, self).__init__(name, init_config, agentConfig, instances=generic_instances)
+
+    def check(self, instance):
+        endpoint = instance.get('prometheus_endpoint')
+        scraper_config = self.config_map[endpoint]
+        self.process(scraper_config, metric_transformers=self.METRIC_TRANSFORMERS)
+
+    def create_generic_instances(self, instances):
+        """
+        Transform each CoreDNS instance into a OpenMetricsBaseCheck instance
+        """
+        generic_instances = []
+        for instance in instances:
+            transformed_instance = self._create_core_dns_instance(instance)
+            generic_instances.append(transformed_instance)
+
+        return generic_instances
+
+    def _create_core_dns_instance(self, instance):
+        """
+        Set up coredns instance so it can be used in OpenMetricsBaseCheck
+        """
+        endpoint = instance.get('prometheus_endpoint')
+        if endpoint is None:
+            raise CheckException("Unable to find prometheus endpoint in config file.")
+        # # coredns uses 'prometheus_endpoint' and not 'prometheus_url', so we have to rename the key
+        # instance['prometheus_url'] = instance.get('prometheus_endpoint', None)
+
+        instance.update({
+            'namespace': 'coredns',
+            'prometheus_url': endpoint,
+            'metrics': [{
+                # https://coredns.io/plugins/metrics/
+                'coredns_dns_request_duration_seconds': 'request_duration.seconds',
+                'coredns_dns_request_size_bytes': 'request_size.bytes',
+                # https://coredns.io/plugins/proxy/
+                'coredns_proxy_request_duration_seconds': 'proxy_request_duration.seconds',
+                # https://coredns.io/plugins/cache/
+                'coredns_cache_size': 'cache_size.count',
+            }]
+        })
+
+        return instance
+
+    def submit_as_gauge_and_monotonic_count(self, metric_suffix, message, scraper_config):
```

Comment:
I think these should only be submitted as `monotonic_count`. See https://github.com/DataDog/integrations-core/issues/1303

Reply:
cc @mfpierre to verify

Reply:
I agree we should use the `send_monotonic_counter` to `True` here I double-checked all the affected metrics are of the `counter` type.
As a result we can remove all the custom functions and `METRIC_TRANSFORMERS`